src/macro/eval/evalMain.ml

```diff
@@ -436,8 +439,52 @@ let rec value_to_expr v p =
 				let args = List.map (fun v -> value_to_expr v p) (Array.to_list e.eargs) in
 				(ECall (epath, args), p)
 		end
-
-	| _ -> exc_string ("Cannot convert " ^ (value_string v) ^ " to expr")
+	| VInstance {ikind = IStringMap smap} ->
+		begin
+			let decl = ref [] in
+			let last_value = ref None in
+			try
+				StringHashtbl.iter (fun key (_,value) ->
+					last_value := Some(value);
+					decl := (EBinop(OpArrow, (EConst (String key),p), (value_to_expr value p)), p) :: !decl
+				) smap;
+				(EArrayDecl(!decl),p)
+			with _ ->
+				match !last_value with
+				| None -> convert_error_message v
+				| Some vv -> convert_map_error_message vv v
+		end
+	| VInstance {ikind = IIntMap imap} ->
+		begin
+			let decl = ref [] in
+			let last_value = ref None in
+			try
+				IntHashtbl.iter (fun key value ->
+					last_value := Some(value);
+					decl := (EBinop(OpArrow, (EConst (Int "0"),p), (value_to_expr value p)), p) :: !decl
```

Comment:
`"0"`?

Reply:
oh my bad xD that was just for debug @Simn